File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/notice/service/dto/NoticeDto.java
package cc.nobrain.dev.userserver.domain.notice.service.dto;

import cc.nobrain.dev.userserver.domain.base.dto.BaseDto;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;

/**
 * DTO for {@link cc.nobrain.dev.userserver.domain.notice.entity.Notice}
 */
@Data
@Accessors(chain = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class NoticeDto extends BaseDto implements Serializable {
    private Long id;
    @NotNull
    private String title;
    private String content;
    private Boolean sticky;
}

File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/notice/service/dto/NoticeDto.kt
package cc.nobrain.dev.userserver.domain.notice.service.dto

import cc.nobrain.dev.userserver.domain.base.dto.BaseDto
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.NotNull
import java.io.Serializable

/**
 *
 */

open class NoticeDto(
    var id: Long,
    @NotNull var title: String,
    var content: String?,
    var sticky: Boolean?
) : BaseDto(), Serializable